pkgs/development/tools/jq/default.nix

```
@@ -1,44 +1,98 @@
+# (enabling Valgrind checks for testing severly increases build time)
+, enableValgrindChecks ? false
+, less ? null, valgrind ? null, which ? null
+}:
+
+assert enableValgrindChecks -> valgrind != null;
+assert enableValgrindChecks -> which != null;
+
+let
+  inherit (stdenv) lib;
+
+  # from ./docs/Gemfile{,.lock}
+  docsRubyApp = rec {
+    pname = "rake";
+    gemdir = ./.;
+    exes = [ "rake" ];
+  };
+  buildDocsRuby = buildPackages.bundlerApp docsRubyApp;
+  docsRuby = bundlerApp docsRubyApp;
+in
 
 stdenv.mkDerivation rec {
   pname = "jq";
   version = "1.6";
 
-  src = fetchurl {
-    url =
-      "https://github.com/stedolan/jq/releases/download/jq-${version}/jq-${version}.tar.gz";
-    sha256 = "0wmapfskhzfwranf6515nzmm84r7kwljgfs7dg6bjgxakbicis2x";
+  src = fetchFromGitHub {
+    owner = "stedolan";
+    repo = "jq";
+    rev = "jq-${version}";
+    sha256 = "1fnf2m46ya7r7afkcb8ba2j0sc4a85m749sh9jz64g4hx6z3r088";
   };
 
   outputs = [ "bin" "doc" "man" "dev" "lib" "out" ];
 
-  buildInputs = [ oniguruma ];
+  nativeBuildInputs = [
+    buildPackages.autoreconfHook
+  ];
+  buildInputs = [
+    oniguruma
+  ];
+  installCheckInputs = [
+    docsRuby
+  ] ++ lib.optionals enableValgrindChecks [
+    valgrind
+    which
+  ];
+
+  patches = [
+    ./nix-docs-ruby.patch
+    ./nix-version.patch
+  ];
+
+  postPatch = ''
+    for file in configure.ac scripts/version; do
+      substituteInPlace "$file" \
+        --subst-var version
+    done
+
+    substituteInPlace Makefile.am \
+      --subst-var-by docs_rake ${lib.escapeShellArg buildDocsRuby}/bin/rake
+  '';
 
   configureFlags = [
+    (lib.enableFeature enableValgrindChecks "valgrind")
     "--bindir=\${bin}/bin"
     "--sbindir=\${bin}/bin"
     "--datadir=\${doc}/share"
     "--mandir=\${man}/share/man"
-  ]
-  # jq is linked to libjq:
-    ++ stdenv.lib.optional (!stdenv.isDarwin) "LDFLAGS=-Wl,-rpath,\\\${libdir}";
+  ] ++ lib.optionals (!stdenv.isDarwin) [
+    # jq is linked to libjq:
+    "LDFLAGS=-Wl,-rpath,\\\${libdir}"
+  ];
 
   doInstallCheck = true;
   installCheckTarget = "check";
 
   postInstallCheck = ''
-    $bin/bin/jq --help >/dev/null
+    ''${bin}/bin/jq --help >/dev/null
   '';
 
   passthru.tests = { inherit (nixosTests) jq; };
+  passthru.updateScript = bundlerUpdateScript "jq";
 
-  meta = with stdenv.lib; {
-    description = "A lightweight and flexible command-line JSON processor";
-    license = licenses.mit;
-    maintainers = with maintainers; [ raskin globin ];
-    platforms = with platforms; linux ++ darwin;
-    downloadPage = "http://stedolan.github.io/jq/download/";
+  meta = {
+    description = ''A lightweight and flexible command-line JSON processor'';
+    downloadPage = "https://stedolan.github.io/jq/download/";
+    homepage = "https://stedolan.github.io/jq/";
+    license = lib.licenses.mit;
+    maintainers = let m = lib.maintainers; in [ m.raskin m.globin m.bb010g ];
+    platforms = let p = lib.platforms; in p.linux ++ p.darwin;
```

```suggestion
    maintainers = with lib.maintainers; [ raskin globin bb010g ];
    platforms = with lib.platforms; linux ++ darwin;
```
Why are you not doing this?

---

Old pattern, sorry, I'll fix that up.